authenticate.go

```diff
@@ -155,6 +162,52 @@ func (c *OktaClient) handleMFAChallenge(transaction api.AuthenticationTransactio
 	case factors.FactorTypeU2F:
 		return c.handleFactorTypeU2F(transaction)
 
+	case factors.FactorTypeWebAuthN:
+		profile, ok := transaction.Embedded.Factor.Profile.(api.FactorProfileWebAuthN)
+		if !ok {
+			c.log("Profile was not of type FactorProfileWebAuthN: %s", transaction.Embedded.Factor.Profile)
+			return c.cancelCurrentFactorWithErrorMessage(transaction, unexpectedErrorMessage)
+		}
+
+		// Setup a context with the timeout set to the value provided by Okta
+		timeoutSeconds := 30
+		ctx, _ := context.WithTimeout(context.Background(), time.Second*time.Duration(timeoutSeconds))
+
+		authResp, err := c.prompts.VerifyU2F(ctx, VerifyU2FRequest{
+			Facet:     c.domain,
+			AppId:     c.domain,
+			KeyHandle: profile.CredentialId,
+			Challenge: transaction.Embedded.Factor.Embedded.Challenge.Challenge,
+		})
+		if err != nil {
+			c.prompts.PresentUserError(fmt.Sprintf("Failed to authenticate: %s\n", err))
+			return c.cancelCurrentFactor(transaction)
+		}
+
+		authData := make([]byte, 37)
+		rpid := sha256.Sum256([]byte(c.domain))
+		copy(authData[0:32], rpid[0:32])
+		authData[32] = 0x80
```

Comment:
When logging into Okta through a browser, this is set to 0x05, which sets the user verified and user present flags.

https://w3c.github.io/webauthn/#authenticator-data